core/src/main/java/org/vertexium/ElementBase.java

```diff
@@ -45,25 +65,10 @@ public Property getProperty(String name, Visibility visibility) {
 
     @Override
     public Iterable<Property> getProperties(String name) {
-        if (ID_PROPERTY_NAME.equals(name)) {
-            ArrayList<Property> result = new ArrayList<>();
-            result.add(getIdProperty());
-            return result;
-        } else if (Edge.LABEL_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            ArrayList<Property> result = new ArrayList<>();
-            result.add(getEdgeLabelProperty());
-            return result;
-        } else if (Edge.OUT_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            ArrayList<Property> result = new ArrayList<>();
-            result.add(getOutVertexIdProperty());
-            return result;
-        } else if (Edge.IN_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
-            ArrayList<Property> result = new ArrayList<>();
-            result.add(getInVertexIdProperty());
-            return result;
-        } else if (Edge.IN_OR_OUT_VERTEX_ID_PROPERTY_NAME.equals(name) && this instanceof Edge) {
+        Property reservedProperty = getReservedProperty(name);
```

Comment:
Do you need to do this? It looks like the implementation of this method just be `return getProperties(null, name)`?